feature(Stock Received and Stock Purchased)

lomamech opened the pull request:
- Implementation of the report allowing analysis of the stocks ordered
  from the stocks received

- Implementing bhPurchaseOrderSelect

closes #3988

lomamech committed:
Sanitaze code: Remove trailing space

mbayopanda commented on client/src/i18n/fr/tree.json:
```diff
@@ -107,6 +107,7 @@
     "PROJECT":"Projets",
     "PURCHASING":"Achat",
     "PURCHASE_ORDER":"Achats",
+    "PURCHASE_ORDER_ANALYSIS": "[Stock] Analyse de commande d'achat",
```
👍

mbayopanda commented on server/controllers/stock/reports/purchaseOrderAnalysis/report.handlebars:
```diff
@@ -0,0 +1,146 @@
+                  <td class="text-bold" colspan="2"> {{ quantity_inStock }} </td>
+                </tr>
+                <tr>
+                  <td class="text-uppercase text-right text-bold" colspan="6">
```
You can also remove this `text-uppercase` class.

mbayopanda commented on server/controllers/stock/reports/purchaseOrderAnalysis/report.handlebars:
```diff
@@ -0,0 +1,146 @@
+                  <td class="text-bold" colspan="2"> {{ quantity_ordered }} </td>
+                </tr>
+                <tr>
+                  <td class="text-uppercase text-right text-bold" colspan="6">
```
You can also remove this `text-uppercase` class.

mbayopanda commented on server/controllers/stock/reports/purchaseOrderAnalysis/report.handlebars:
```diff
@@ -0,0 +1,146 @@
+                  <td class="text-bold" colspan="2"> {{ quantity_difference }} </td>
+                </tr>
+                <tr>
+                  <td class="text-uppercase text-right text-bold" colspan="6">
```
You can also remove this `text-uppercase` class.

mbayopanda commented on server/controllers/stock/reports/purchaseOrderAnalysis/report.handlebars:
```diff
@@ -0,0 +1,146 @@
+              <th width="12%" class="text-uppercase"> {{translate 'FORM.LABELS.QUANTITY_ENTRY'}} </th>
+              <th width="15%" class="text-uppercase"> {{translate 'FORM.LABELS.QUANTITY_DIFFERENCE'}} </th>
+              <th width="17%" class="text-uppercase" colspan="2"> {{translate 'FORM.LABELS.STATUS'}} </th>
+              <th width="18%" class="text-uppercase"> {{translate 'FORM.LABELS.LAST_ENTRY_DATE'}} </th>
```
Could you remove all these `text-uppercase` classes.

mbayopanda commented on package.json:
```diff
@@ -8,9 +8,9 @@
-    "test:server-unit": "cross-env SUITE=\"server-unit\" ./sh/test.sh",
+    "test:ends": "SUITE=\"end-to-end\" ./sh/test.sh",
+    "test:client-unit": "SUITE=\"client-unit\" ./sh/test.sh",
+    "test:server-unit": "SUITE=\"server-unit\" ./sh/test.sh",
```
Could you restore `cross-env` command, such as : 
```json
"test:ends": "cross-env SUITE=\"end-to-end\" ./sh/test.sh",
"test:client-unit": "cross-env SUITE=\"client-unit\" ./sh/test.sh",
"test:server-unit": "cross-env SUITE=\"server-unit\" ./sh/test.sh",
```

mbayopanda reviewed (commented):
@lomamech this is a good job, the report is very nice.

I made just some comments.

lomamech committed:
improvement(Stock Received and Stock Purchased)
- Restaure cross-end in package.json
- Use class text-capitalize instead text-uppercase

lomamech committed:
Resolve conflict
